### DISCUSSION TITLE
Stack corruption when using stand-alone wasm in a worker

### DISCUSSION THREAD

**Discussion by 98ahni** [emscripten-core/emscripten#22333] [Q&A]
I want to use web assembly in a worker to do some heavy calculations in the background while the rest of the app runs like normal. However when it runs the main loop it runs once and then aborts with this message: `Aborted(Stack overflow! Stack cookie has been overwritten at 0x00000004, expected hex dwords 0x89BACDFE and 0x2135467, but received 0x00000000 0x00000000)`. I have no idea what causes it so I don't know what details would be relevant to add. The same code can run without error in the main program but is too slow.   
  
### What have I tried?  
- Building with different levels of compiler optimizations  
- Increasing the stack size  
- Setting `-fsanitize=address`  

None of which had any effect on the error.
### Why do I not use a wasm-worker/pthread?  
The service I use for hosting doesn't allow me to set my own http headers and thus locking me out of having access to SharedArrayBuffer, which they need to run.

**Comment by sbc100** [comment]
Emscripten-generated module should run just fine in web workers.  

- Can you reproduce the issue locally?  
- Does it happen only when run run in a worker?   i.e. does your program work fine when run on the main thread?
- Can you share details of how you start worker and load the emscripten module?
- Can you share the full set of link flags you used?

**Comment by 98ahni** [comment]
While adding the functions back to the main program I noticed that I took the address of an std::vector's data and then resized it, causing the pointer to be stale. I must have been lucky every time I ran it until I wasn't.  
  
I'm sorry for the false alarm.
